src/ResourceManager/Sql/Commands.Sql/Database Backup/Services/AzureSqlDatabaseBackupCommunicator.cs

```diff
@@ -400,11 +396,9 @@ private Management.Sql.LegacySdk.SqlManagementClient GetLegacySqlClient()
         private Management.Sql.SqlManagementClient GetCurrentSqlClient()
         {
             // Get the SQL management client for the current subscription
-            if (SqlClient == null)
-            {
-                SqlClient = AzureSession.Instance.ClientFactory.CreateArmClient<Management.Sql.SqlManagementClient>(Context, AzureEnvironment.Endpoint.ResourceManager);
-            }
-            return SqlClient;
+            // Note: client is not cached in static field because that causes ObjectDisposedException in functional tests.
+            var sqlClient = AzureSession.Instance.ClientFactory.CreateArmClient<Management.Sql.SqlManagementClient>(Context, AzureEnvironment.Endpoint.ResourceManager);
```

Comment:
@dealaus nit: no need for `sqlClient` when you can just return the result of the method 😀 

@dealaus @jaredmoo also, out of curiosity, what kind of caching issue was causing the previous client issues? And since we're now instantiating a new client each time a cmdlet that uses this communicator makes a call to a non-legacy SDK type, will there be any performance issues, or anything that your team would be concerned about?

Reply:
The cached SqlClient was not refreshed when constructing a AzureSqlDatabaseBackupCommunicator with new subscription id. 

We could handle this similar to what the LegacyClient does, but 

```
            if (context.Subscription != Subscription)
            {
                Subscription = context.Subscription;
                LegacyClient = null;
            }
```

but when I chatted with @jaredmoo about this in May when we first discovered it, we agreed to just remove the mutable static object.

Reply:
Got it. Thanks for clarifying.